### DISCUSSION TITLE
Choice of Unit Testing Framework

### DISCUSSION THREAD

**Discussion by Structed** [Annonator/PlayFabBuddy#51] [General]
Currently, MS Test is being used for Unit Tests.
However, this is not a very broadly used testing framework in the public - and even Microsoft uses [xUnit](https://xunit.net/) for APS.NET and otehr projects built by the .NET Team.

Therefore, and since there was no massive investment in MS TEST, I would like to propose to switch to xUnit.

xUnit is:
* More widely used
** Translates to more experience in the larger community
* Data driven tests are very easy to do
* open source
* part of the .NET Foundation
* very extensible
* Mocking with [Moq](https://github.com/moq/moq4) is a charm - it's a massive pain with MS Test.

**Comment by Annonator** [comment]
Happy to have this discussion as initially I just "right clicked add test" in VS and did not put any thought into it. 
That said I would love to have this discussion around the pros and cons of the frameworks as a learning experience, maybe even extend the scope to NUnit.

As far as MSTest goes, it is open source (https://github.com/microsoft/testfx) and you could use Moq in MS test tests. So, is MS test really that far behind NUnit / XUnit? 

PS: Yes, I do work for this MS company thingy, so don't get me wrong, not trying to defend MS test here, really want to use this as learning opportunity as I don't know enough about the topic today.
